src/appleseed-max-impl/appleseedrenderer/projectbuilder.cpp

```diff
@@ -810,16 +819,17 @@ namespace
             const int ObjectMotionBlur = 1;
             if (node->GetMotBlurOnOff(time) && node->MotBlur() == ObjectMotionBlur)
             {
-                Control* tm_controller = node->GetTMController();
-
-                const bool is_animated =
-                    tm_controller->GetPositionController()->IsAnimated() > 0 ||
-                    tm_controller->GetRotationController()->IsAnimated() > 0 ||
-                    tm_controller->GetScaleController()->IsAnimated() > 0;
+                bool is_animated = is_node_animated(node);
+                INode* parent_node = node->GetParentNode();
+                while (!is_animated && !parent_node->IsRootNode())
```

Comment:
I think there's a bug here, you'll never check whether the root node is animated or not.

Here's a solution without any duplication that properly checks the root node:

    bool is_animated = false;

    while (!is_animated)
    {
        is_animated = is_node_animated(node);

        if (node->IsRootNode())
            break;

        node = node->GetParentNode();
    }

Reply:
Hi @dictoon,
I didn't want to overwrite `node` because it's animation is taken later.
Also I don't think root node can be animated..

Reply:
Alright.